Mathlib/Tactic/Abel.lean

```diff
@@ -473,7 +473,7 @@ open Elab.Tactic Parser.Tactic
 /-- Use `abel_nf` to rewrite the main goal. -/
 def abelNFTarget (s : IO.Ref AtomM.State) (cfg : AbelNF.Config) : TacticM Unit := withMainContext do
   let goal ← getMainGoal
-  let tgt ← instantiateMVars (← goal.getType)
+  let tgt ← instantiateMVars (← goal.getType')
```

Comment:
I think this is too strong (it does whnf at default transparency).

I'm not sure about doing reduction (I know @fpvandoorn is not sold on doing whnf before matching). There's also `cleanupAnnotations (← instatiateMVars (← goal.getType))`.

Reply:
In the following code snippet, I would indeed hope that `abel_nf` doesn't unfold the definition of `MyTrue`.
```
def MyTrue := True
example : MyTrue := by
  abel_nf
```

Reply:
It should unfold it if it is `@[reducible]` though, right?

Reply:
Would `← whnfR (← instantiateMVars (← goal.getType'))` and then guard-messaging the test suggested by Floris as an error be a solution with which everyone is happy?

Reply:
Watch out, whnfR has to occur before instantiateMVars.

Better I think would be withReducible around the current code. (With getType')

Reply:
Kyle, assuming that
```lean
  let tgt ← Lean.Meta.withReducible (instantiateMVars (← goal.getType'))
```
is the change that you suggested, then Floris' example actually gets solved by `abel_nf`.

Reply:
Sorry, I made a mistake with what I wrote, I wrote it on my phone. You need to be careful with `withReducible` and `←` lifting. The `getType'` function doesn't need `instantiateMVars`, so with that simplification, the correct version should be
```lean
  let tgt ← withReducible goal.getType'
```

Reply:
Thanks Kyle: I used your latest version and it passes the tests, both successfully and not!